django/forms/widgets.py

```diff
@@ -573,6 +573,24 @@ def render_options(self, selected_choices):
                 output.append(self.render_option(selected_choices, option_value, option_label))
         return '\n'.join(output)
 
+    def option_has_empty_value(self, choice):
+        choice_value = choice[0]
+        if isinstance(choice_value, six.string_types):
+            return len(choice_value) == 0
+        return choice_value is None
+
+    def use_required_attribute(self, initial):
+        # If first item has a value, rendering required attribute causes HTML validation error.
+        # More info: https://github.com/validator/validator/issues/47#issuecomment-77534952
```

Comment:
Your first comment is quite explicit, I don't think the __more info__ part is required.